ballerina-shell/README.md

```diff
@@ -0,0 +1,130 @@
+# Ballerina Shell
+
+A REPL program for the [ballerina language](https://github.com/ballerina-platform/ballerina-lang). Ballerina is an open source programming language and platform for cloud-era application programmers to easily write software that just works.
+
+The Ballerina-shell tool is an interactive tool for learning the Ballerina programming language and prototyping Ballerina code. Ballerina-shell is a Read-Evaluate-Print Loop (REPL), which evaluates declarations, statements, and expressions as they are entered and immediately shows the results. Currently, the tool is run via the command line. Using Ballerina-shell, you can enter program statements one at a time and immediately see the result.
+
+## Demo
+
+![Recording](./docs/demo.gif)
+
+## Modules
+
+The project is implemented in two base modules.
+
+- **shell-rt** - Module including runtime dependencies for ballerina programs generated. You may find the source code for this
+  module [here](shell-rt).
+- **shell-core** - Module including all the base evaluation classes. This has all the base components to evaluate and run a
+  string. All other components are built on top of this module. You may find the source code for this
+  module [here](shell-core).
+- **shell-cli** - A command-line interface built on top of shell. Includes multi-line inputs, color-coded outputs,
+  keyword-based auto-completion, etc... You may find the source code for this module [here](shell-cli).
+
+## Known Issues
+
+- **Assignments to global variables in closures or class methods will not work** - Assignments done to global variables
+  in closures will not be reflected after the execution. The changes will be visible only for the scope belonging to the
+  snippet where the closure was defined.
+
+  ```ballerina
+  int x = 10
+  var f = function () { x = 12; }
+  f()
+  x   // <- this should output 12 but will output 10 instead
+  ```
+
+- **Enum definition with explicit expression is not fully supported** - Enum declarations with explicit expressions are
+  not supported fully. Some functionalities may not work properly. Following is an example.
+
+  ```ballerina
+  // This will not work due to type EN converting to english
+  enum Language {EN="english", TA="tamil", SI="sinhala"}
+  EN variable = "english"  // <- Error
+  ```
+
+- **Support with external libraries not tested completely** - Even though it is possible to run this against external libraries, this is not fully tested. Some issues, especially relating to invincible types, may occur. For an example following is a possible problematic scenario.
+
+  ```ballerina
+  import abc/pqr
+  // If pqr:call() returns a value of which the type is private, following will fail.
```

Comment:
Since we don't have private types shall we say
```suggestion
  // If pqr:call() returns a value of which the type is not public, following will fail.
```